cli/src/main/kotlin/commands/ReporterCommand.kt

```diff
@@ -252,43 +254,41 @@ class ReporterCommand : CliktCommand(
             reportSpecificOptionsMap[option.first] = option.second
         }
 
-        var failure = false
-        val reportDurationMap = mutableMapOf<Reporter, TimedValue<List<File>>>()
+        val reportDurationMap = measureTimedValue {
+            runBlocking(Dispatchers.Default) {
+                reportFormats.map { reporter ->
+                    async {
+                        val threadName = Thread.currentThread().name
+                        println("Generating the '${reporter.reporterName}' report in thread '$threadName'...")
+
+                        reporter to measureTimedValue {
+                            val options = reportOptionsMap[reporter.reporterName.toUpperCase()].orEmpty()
+                            runCatching { reporter.generateReport(input, outputDir, options) }
+                        }
+                    }
+                }.awaitAll()
+            }
+        }
 
-        reportFormats.forEach { reporter ->
-            val options = reportOptionsMap[reporter.reporterName.toUpperCase()].orEmpty()
+        var failureCount = 0
 
-            println("Creating the '${reporter.reporterName}' report...")
+        reportDurationMap.value.forEach { (reporter, timedValue) ->
+            val name = reporter.reporterName
 
-            @Suppress("TooGenericExceptionCaught")
-            try {
-                reportDurationMap[reporter] = measureTimedValue {
-                    reporter.generateReport(input, outputDir, options)
-                }
-            } catch (e: Exception) {
+            timedValue.value.onSuccess { files ->
+                println("Successfully created the '$name' report at $files in ${timedValue.duration.inSeconds}s.")
+            }.onFailure { e ->
                 e.showStackTrace()
 
-                log.error { "Could not create '${reporter.reporterName}' report: ${e.collectMessagesAsString()}" }
+                log.error { "Could not create '$name' report: ${e.collectMessagesAsString()}" }
```

Comment:
Since you already edit this you could also add the duration to this log message.

Reply:
Does it really make sense to log the duration in an error case? I mean, it would not be very telling, as the report could be incomplete.

Reply:
In case you have an unexpectedly long total duration and multiple reports failed, it would be interesting to know if one the failed reports is the culprit.

Reply:
Done.